### DISCUSSION TITLE
is it possible to set max log length?

### DISCUSSION THREAD

**Discussion by seaguest** [uber-go/zap#1269] [General]
Hello,

we log debug information with some objects, sometimes the log could be very long (more than 10K chars), is it possible to set a max log length, truncate the message if it is too long?

**Comment by JacobOaks** [comment]
I think you can do this fairly easily with a custom implementation of `zapcore.Core` that truncates messages in its `Write` method. It can just wrap another core. Something like:

```go
// logLimitCore truncates messages longer than its limit field.
type logLimitCore struct {
    underlying zapcore.Core
    limit      int
}

var _ zapcore.Core = logLimitCore{}

func (c logLimitCore) Enabled(l zapcore.Level) bool {
    return c.underlying.Enabled(l)
}

func (c logLimitCore) With(fields []zapcore.Field) zapcore.Core {
    return logLimitCore{
        underlying: c.underlying.With(fields),
        limit:      c.limit,
    }
}

func (c logLimitCore) Check(ent zapcore.Entry, ce *zapcore.CheckedEntry) *zapcore.CheckedEntry {
	if c.Enabled(ent.Level) {
		return ce.AddCore(ent, c)
	}
	return ce
}

func (c logLimitCore) Write(ent zapcore.Entry, fields []zapcore.Field) error {
    ent.Message = ent.Message[:c.limit] // Truncate messages longer than limit
    return c.underlying.Write(ent, fields)
}

func (c logLimitCore) Sync() error {
    return c.underlying.Sync()
}
```

Then you can wrap around whatever core you're currently using, for example the pre-defined production logger's core:
```go
base, err := zap.NewProduction()
if err != nil {
    return fmt.Errorf("create prod base logger: %w", err)
}
logger := zap.New(logLimitCore{
    underlying: base.Core(),
    limit:      _myLogLimit,
})
```

**Reply by JacobOaks** [reply]
Or what might be better is to use a custom encoder wrapper, similar to how they do it [here](https://stackoverflow.com/questions/73469128/hide-sensitive-fields-in-uber-zap-go/73472303#73472303), where you truncate the messages in the `EncodeEntry` method.
